tests/test_visitors/test_ast/test_classes/test_buggy_super.py

```diff
@@ -0,0 +1,104 @@
+import pytest
+
+from wemake_python_styleguide.violations.oop import BuggySuperContextViolation
+from wemake_python_styleguide.visitors.ast.classes import BuggySuperCallVisitor
+
+error_dict_comprehension = """
+    {super().make_key(it): make_value(it) for it in items}
+"""
+
+error_set_comprehension = """
+    {super().transmute(it) for it in items}
+"""
+
+error_list_comprehension = """
+    [super().transmute(it) for it in items]
+"""
+
+error_generator_expression = """
+    (super().transmute(it) for it in items)
+"""
+
+error_nested_comprehensions = """
+    [
+        it + 1
+        for it in (super().transmute(i) for i in range(10))
+    ]
+"""
+
+error_nested_methods = """
```

Comment:
We are still missing a correct test for this feature.

We need two:
1. One with `super(klass, self)` (if it works in CPython)
2. With regular non-nested `super()` call - just to be sure